crates/uv-requirements/src/source_tree.rs

```diff
@@ -106,7 +106,8 @@ impl<'a, Context: BuildContext> SourceTreeResolver<'a, Context> {
                 origin: Some(origin.clone()),
                 marker: requirement
                     .marker
-                    .and_then(|marker| marker.simplify_extras(extras)),
+                    .map(|marker| marker.simplify_extras(extras))
+                    .filter(|marker| !marker.is_true()),
```

Comment:
nit: `is_false`

Reply:
I think `!marker.is_true()` does not imply `marker.is_false()`. I do like the succinctness of the method names, but they do suggest a mutual exclusivity that isn't there. Perhaps `is_always_true` and `is_always_false` are better names? Not sure.

Reply:
Good call, totally missed that